### ISSUE TITLE
Mint Start Error

### ISSUE THREAD

**Issue by mrclaybarton** [beefproject/beef#1511]
[code]
test-VirtualBox beef # ruby beef
/usr/local/rvm/gems/ruby-2.4.1/gems/bundler-1.16.1/lib/bundler/runtime.rb:84:in `rescue in block (2 levels) in require': There was an error while trying to load the gem 'uglifier'. (Bundler::GemRequireError)
Gem Load Error is: wrong argument type Class (expected Module)
Backtrace for gem load error is:
/usr/local/rvm/gems/ruby-2.4.1/gems/therubyracer-0.12.2/lib/v8/conversion.rb:23:in `include'
/usr/local/rvm/gems/ruby-2.4.1/gems/therubyracer-0.12.2/lib/v8/conversion.rb:23:in `block (2 levels) in <top (required)>'
/usr/local/rvm/gems/ruby-2.4.1/gems/therubyracer-0.12.2/lib/v8/conversion.rb:22:in `class_eval'
/usr/local/rvm/gems/ruby-2.4.1/gems/therubyracer-0.12.2/lib/v8/conversion.rb:22:in `block in <top (required)>'
/usr/local/rvm/gems/ruby-2.4.1/gems/therubyracer-0.12.2/lib/v8/conversion.rb:21:in `each'
/usr/local/rvm/gems/ruby-2.4.1/gems/therubyracer-0.12.2/lib/v8/conversion.rb:21:in `<top (required)>'
/usr/local/rvm/gems/ruby-2.4.1/gems/therubyracer-0.12.2/lib/v8.rb:22:in `require'
/usr/local/rvm/gems/ruby-2.4.1/gems/therubyracer-0.12.2/lib/v8.rb:22:in `<top (required)>'
/usr/local/rvm/gems/ruby-2.4.1/gems/execjs-2.7.0/lib/execjs/ruby_racer_runtime.rb:108:in `require'
/usr/local/rvm/gems/ruby-2.4.1/gems/execjs-2.7.0/lib/execjs/ruby_racer_runtime.rb:108:in `available?'
/usr/local/rvm/gems/ruby-2.4.1/gems/execjs-2.7.0/lib/execjs/runtimes.rb:63:in `each'
/usr/local/rvm/gems/ruby-2.4.1/gems/execjs-2.7.0/lib/execjs/runtimes.rb:63:in `find'
/usr/local/rvm/gems/ruby-2.4.1/gems/execjs-2.7.0/lib/execjs/runtimes.rb:63:in `best_available'
/usr/local/rvm/gems/ruby-2.4.1/gems/execjs-2.7.0/lib/execjs/runtimes.rb:57:in `autodetect'
/usr/local/rvm/gems/ruby-2.4.1/gems/execjs-2.7.0/lib/execjs.rb:5:in `<module:ExecJS>'
/usr/local/rvm/gems/ruby-2.4.1/gems/execjs-2.7.0/lib/execjs.rb:4:in `<top (required)>'
/usr/local/rvm/gems/ruby-2.4.1/gems/uglifier-3.2.0/lib/uglifier.rb:5:in `require'
/usr/local/rvm/gems/ruby-2.4.1/gems/uglifier-3.2.0/lib/uglifier.rb:5:in `<top (required)>'
/usr/local/rvm/gems/ruby-2.4.1/gems/bundler-1.16.1/lib/bundler/runtime.rb:81:in `require'
/usr/local/rvm/gems/ruby-2.4.1/gems/bundler-1.16.1/lib/bundler/runtime.rb:81:in `block (2 levels) in require'
/usr/local/rvm/gems/ruby-2.4.1/gems/bundler-1.16.1/lib/bundler/runtime.rb:76:in `each'
/usr/local/rvm/gems/ruby-2.4.1/gems/bundler-1.16.1/lib/bundler/runtime.rb:76:in `block in require'
/usr/local/rvm/gems/ruby-2.4.1/gems/bundler-1.16.1/lib/bundler/runtime.rb:65:in `each'
/usr/local/rvm/gems/ruby-2.4.1/gems/bundler-1.16.1/lib/bundler/runtime.rb:65:in `require'
/usr/local/rvm/gems/ruby-2.4.1/gems/bundler-1.16.1/lib/bundler.rb:114:in `require'
/usr/beef/core/loader.rb:10:in `<top (required)>'
/usr/local/rvm/rubies/ruby-2.4.1/lib/ruby/site_ruby/2.4.0/rubygems/core_ext/kernel_require.rb:55:in `require'
/usr/local/rvm/rubies/ruby-2.4.1/lib/ruby/site_ruby/2.4.0/rubygems/core_ext/kernel_require.rb:55:in `require'
beef:44:in `<main>'
Bundler Error Backtrace:
	from /usr/local/rvm/gems/ruby-2.4.1/gems/bundler-1.16.1/lib/bundler/runtime.rb:80:in `block (2 levels) in require'
	from /usr/local/rvm/gems/ruby-2.4.1/gems/bundler-1.16.1/lib/bundler/runtime.rb:76:in `each'
	from /usr/local/rvm/gems/ruby-2.4.1/gems/bundler-1.16.1/lib/bundler/runtime.rb:76:in `block in require'
	from /usr/local/rvm/gems/ruby-2.4.1/gems/bundler-1.16.1/lib/bundler/runtime.rb:65:in `each'
	from /usr/local/rvm/gems/ruby-2.4.1/gems/bundler-1.16.1/lib/bundler/runtime.rb:65:in `require'
	from /usr/local/rvm/gems/ruby-2.4.1/gems/bundler-1.16.1/lib/bundler.rb:114:in `require'
	from /usr/beef/core/loader.rb:10:in `<top (required)>'
	from /usr/local/rvm/rubies/ruby-2.4.1/lib/ruby/site_ruby/2.4.0/rubygems/core_ext/kernel_require.rb:55:in `require'
	from /usr/local/rvm/rubies/ruby-2.4.1/lib/ruby/site_ruby/2.4.0/rubygems/core_ext/kernel_require.rb:55:in `require'
	from beef:44:in `<main>'
[/code]

rvm install 2.3.0

[code]
test-VirtualBox beef # rvm install 2.3.0
ruby-2.3.0 - #removing src/ruby-2.3.0..
Searching for binary rubies, this might take some time.
No binary rubies available for: mint/18.3/x86_64/ruby-2.3.0.
Continuing with compilation. Please read 'rvm help mount' to get more information on binary rubies.
Checking requirements for mint.
Requirements installation successful.
Installing Ruby from source to: /usr/local/rvm/rubies/ruby-2.3.0, this may take a while depending on your cpu(s)...
ruby-2.3.0 - #downloading ruby-2.3.0, this may take a while depending on your connection...
ruby-2.3.0 - #extracting ruby-2.3.0 to /usr/local/rvm/src/ruby-2.3.0....
ruby-2.3.0 - #applying patch /usr/local/rvm/patches/ruby/ruby_2_3_gcc7.patch.
ruby-2.3.0 - #applying patch /usr/local/rvm/patches/ruby/changeset_r53419.diff.
ruby-2.3.0 - #applying patch /usr/local/rvm/patches/ruby/2.3.0/random_c_using_NR_prefix.patch.
ruby-2.3.0 - #configuring..........................................................
ruby-2.3.0 - #post-configuration..
ruby-2.3.0 - #compiling........................................................................
Error running '__rvm_make -j1',
please read /usr/local/rvm/log/1518892307_ruby-2.3.0/make.log
There has been an error while running make. Halting the installation.
[/code]

make.log

[code]
ossl.c:507:1: warning: ‘ossl_dyn_destroy_callback’ defined but not used [-Wunused-function]
 ossl_dyn_destroy_callback(struct CRYPTO_dynlock_value *l, const char *file, int line)
 ^
ossl.c:520:22: warning: ‘ossl_thread_id’ defined but not used [-Wunused-function]
 static unsigned long ossl_thread_id(void)
                      ^
Makefile:301: recipe for target 'ossl.o' failed
make[2]: *** [ossl.o] Error 1
make[2]: Leaving directory '/usr/local/rvm/src/ruby-2.3.0/ext/openssl'
exts.mk:207: recipe for target 'ext/openssl/all' failed
make[1]: *** [ext/openssl/all] Error 2
make[1]: Leaving directory '/usr/local/rvm/src/ruby-2.3.0'
uncommon.mk:203: recipe for target 'build-ext' failed
make: *** [build-ext] Error 2
++ return 2
[/code]

**Comment by mrclaybarton** [comment]
test@test-VirtualBox ~ $  inxi -S
System:    Host: test-VirtualBox Kernel: 4.10.0-38-generic x86_64 (64 bit)
           Desktop: MATE 1.18.0  Distro: Linux Mint 18.3 Sylvia

**Comment by bcoles** [comment]
Does the patch in the comment at the end of [this thread](https://github.com/beefproject/beef/issues/988) work for you?

**Comment by bcoles** [comment]
Hi @mrclaybarton 

If it helps, here's my terminal output for installing and running BeEF on Linux Mint 18 with ruby 2.3.1p112.

```
user@mint-18 ~/Desktop $ git clone https://github.com/beefproject/beef
Cloning into 'beef'...
remote: Counting objects: 34798, done.
remote: Compressing objects: 100% (7/7), done.
remote: Total 34798 (delta 2), reused 5 (delta 1), pack-reused 34790
Receiving objects: 100% (34798/34798), 11.01 MiB | 948.00 KiB/s, done.
Resolving deltas: 100% (21805/21805), done.
Checking connectivity... done.
user@mint-18 ~/Desktop $ cd beef/
user@mint-18 ~/Desktop/beef $ git status
On branch master
Your branch is up-to-date with 'origin/master'.
nothing to commit, working directory clean
user@mint-18 ~/Desktop/beef $ su
Password: 
mint-18 beef # 
mint-18 beef # ./install 
,                              
                                     lkOl                             
                                     od cOc                           
                                     'X,  cOo.                        
                                      cX,   ,dkc.                     
                                       ;Kd.    ,odo,.                 
                                        .dXl   .  .:xkl'              
                                          'OKc  .;c'  ,oOk:           
                                            ,kKo. .cOkc. .lOk:.       
                                              .dXx.  :KWKo. 'dXd.     
                                                .oXx.  cXWW0c..dXd.   
                                                  oW0   .OWWWNd.'KK.  
                                          ....,;lkNWx     KWWWWX:'XK. 
 ,o:,                          .,:odkO00XNK0Okxdlc,.     .KWWWWWWddWd 
  K::Ol                   .:d0NXK0OkxdoxO'             .lXWWWWWWWWKW0 
  od  d0.              .l0NKOxdooooooox0.        .,cdOXWWWWWWWWWWWWWx 
  :O   ;K;           ;kN0kooooooooooooK:  .':ok0NWWWWWWWWWWWWWWWWWWK. 
  'X    .Kl        ;KNOdooooooooooooooXkkXWWWWWWWWWWWWWWWWWWWWWWWNd.  
  .N. o. .Kl     'OW0doooooooooooooodkXWWWWWWWWWWWWWWWWWWWWWWWW0l.    
   0l oK' .kO:';kNNkoooooooooooook0XWWWWWWWWWWWWWWWWWWWWWWWKx:.       
   lX.,WN:  .:c:xWkoooooooooood0NWW0OWWWWWWWWWWWWWWWWWWWKo.           
    0O.0WWk'   .XKoooooooooooONWWNo  dWWWWWWWWWWWWWWWWWl              
     oKkNWWWX00NWXdooooooooxXWWNk'   dWWWWWWWWWWWWWWWWX               
      .cONWWWWWWWWOoooooooONWWK:...c0WWWWWWWWWWWWWWWWWW:              
         .;oONWWWWxooooodKWWWWWWWWWWWWWWWWWWWWWWWWWWWWWX.             
              'XW0oooookNWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWd             
              oW0ooooo0WWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWO             
             ;NXdooodKWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWx             
          ;xkOOdooooxOO0KNWWWWWWWWWWWWWWWWWWWWWWWWWWWWWX.             
         .NOoddxkkkkxxdoookKWWWWWWWWWWWWWWWWWWWWWWWWWWX'              
          :KNWWWWWWWWWWX0xooONWWWWWWWWWWWWWWWWWWWWWWWk.               
         .xNXxKWWWWWWWOXWWXxoKWWWWWWWWWWWWWWWWWWWWNk'                 
         OWl cNWWWWWWWk oNWNxKWWWWWWWWWWWWWWWWWNOl.                   
        ,Wk  xWWWWWWWWd  xWWNWWWWWWWWWWWWXOdc,.                       
        .N0   lOXNX0x;  .KWWWWWWWWWWWNkc.                             
         :NO,         'lXWWWWWWWWWNk:.                                
          .dXN0OkxkO0NWWWWWWWWWWKl.                                   
             .';o0WWWWWWWWWWWNk;                                      
                  .cxOKXKKOd;.                                        
                                                                      

#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#
                   -- [ BeEF Installer ] --                      
#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#

[WARN]   This script will install BeEF and its required dependencies (including operating system packages).
Are you sure you wish to continue (Y/n)? 
[INFO]   Detecting OS...
[INFO]   Operating System: Linux
[INFO]   Launching Linux install...
[INFO]   Detecting Linux OS distribution...
[INFO]   OS Distribution: Debian
[INFO]   Installing Debian prerequisite packages...
Hit:1 http://archive.ubuntu.com/ubuntu xenial InRelease                                          
Hit:2 http://archive.ubuntu.com/ubuntu xenial-updates InRelease                                  
Hit:3 http://archive.ubuntu.com/ubuntu xenial-backports InRelease                                
Ign:4 http://packages.linuxmint.com sarah InRelease                                              
Hit:5 http://archive.canonical.com/ubuntu xenial InRelease                                       
Hit:6 http://packages.linuxmint.com sarah Release                                      
Hit:8 http://security.ubuntu.com/ubuntu xenial-security InRelease                      
Reading package lists... Done
Reading package lists... Done
Building dependency tree       
Reading state information... Done
autoconf is already the newest version (2.69-9).
automake is already the newest version (1:1.15-4ubuntu1).
bison is already the newest version (2:3.0.4.dfsg-1).
build-essential is already the newest version (12.1ubuntu2).
libncurses5-dev is already the newest version (6.0+20160213-1ubuntu1).
libreadline6-dev is already the newest version (6.3-8ubuntu2).
libsqlite3-0 is already the newest version (3.11.0-1ubuntu1).
libsqlite3-dev is already the newest version (3.11.0-1ubuntu1).
libtool is already the newest version (2.4.6-0.1).
libyaml-dev is already the newest version (0.1.6-3).
sqlite3 is already the newest version (3.11.0-1ubuntu1).
curl is already the newest version (7.47.0-1ubuntu2.6).
git is already the newest version (1:2.7.4-0ubuntu1.3).
libc6-dev is already the newest version (2.23-0ubuntu10).
libssl-dev is already the newest version (1.0.2g-1ubuntu4.10).
libxml2-dev is already the newest version (2.9.3+dfsg1-1ubuntu0.5).
libxslt1-dev is already the newest version (1.1.28-2.1ubuntu0.1).
openssl is already the newest version (1.0.2g-1ubuntu4.10).
subversion is already the newest version (1.9.3-2ubuntu1.1).
zlib1g is already the newest version (1:1.2.8.dfsg-2ubuntu4.1).
zlib1g-dev is already the newest version (1:1.2.8.dfsg-2ubuntu4.1).
nodejs is already the newest version (4.2.6~dfsg-1ubuntu4.1).
0 to upgrade, 0 to newly install, 0 to remove and 720 not to upgrade.
[INFO]   Detecting Ruby environment...
[INFO]   Ruby version 2.3.1 is installed
[INFO]   Detecting bundler gem...
[INFO]   Installing bundler gem
Fetching: bundler-1.16.1.gem (100%)
Successfully installed bundler-1.16.1
Parsing documentation for bundler-1.16.1
Installing ri documentation for bundler-1.16.1
Done installing documentation for bundler after 4 seconds
1 gem installed
Installing required Ruby gems...
Don't run Bundler as root. Bundler can ask for sudo if it is needed, and installing your bundle as
root will break this application for all non-root users on this machine.
Fetching gem metadata from https://rubygems.org/............
Fetching gem metadata from https://rubygems.org/..
Resolving dependencies...
Fetching rake 12.1.0
Installing rake 12.1.0
Fetching public_suffix 3.0.0
Installing public_suffix 3.0.0
Fetching addressable 2.5.2
Installing addressable 2.5.2
Fetching ansi 1.5.0
Installing ansi 1.5.0
Fetching buftok 0.2.0
Installing buftok 0.2.0
Using bundler 1.16.1
Fetching chunky_png 1.3.8
Installing chunky_png 1.3.8
Fetching daemons 1.2.4
Installing daemons 1.2.4
Fetching data_objects 0.10.17
Installing data_objects 0.10.17
Fetching dm-core 1.2.1
Installing dm-core 1.2.1
Fetching dm-do-adapter 1.2.0
Installing dm-do-adapter 1.2.0
Fetching dm-migrations 1.2.0
Installing dm-migrations 1.2.0
Fetching fastercsv 1.5.5
Installing fastercsv 1.5.5
Fetching json 1.8.6
Installing json 1.8.6 with native extensions
Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

    current directory: /var/lib/gems/2.3.0/gems/json-1.8.6/ext/json/ext/generator
/usr/bin/ruby2.3 -r ./siteconf20180307-8200-8dtot.rb extconf.rb
mkmf.rb can't find header files for ruby at /usr/lib/ruby/include/ruby.h

extconf failed, exit code 1

Gem files will remain installed in /var/lib/gems/2.3.0/gems/json-1.8.6 for inspection.
Results logged to /var/lib/gems/2.3.0/extensions/x86_64-linux/2.3.0/json-1.8.6/gem_make.out

An error occurred while installing json (1.8.6), and Bundler cannot continue.
Make sure that `gem install json -v '1.8.6'` succeeds before bundling.

In Gemfile:
  dm-serializer was resolved to 1.2.2, which depends on
    json
mint-18 beef # apt-get install ruby-dev
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following additional packages will be installed:
  libgmp-dev libgmpxx4ldbl ruby2.3-dev
Suggested packages:
  gmp-doc libgmp10-doc libmpfr-dev
The following NEW packages will be installed:
  libgmp-dev libgmpxx4ldbl ruby-dev ruby2.3-dev
0 to upgrade, 4 to newly install, 0 to remove and 720 not to upgrade.
Need to get 1,360 kB of archives.
After this operation, 6,514 kB of additional disk space will be used.
Do you want to continue? [Y/n] 
Get:1 http://archive.ubuntu.com/ubuntu xenial/main amd64 libgmpxx4ldbl amd64 2:6.1.0+dfsg-2 [8,948 B]
Get:2 http://archive.ubuntu.com/ubuntu xenial/main amd64 libgmp-dev amd64 2:6.1.0+dfsg-2 [314 kB]
Get:3 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 ruby2.3-dev amd64 2.3.1-2~16.04.6 [1,033 kB]
Get:4 http://archive.ubuntu.com/ubuntu xenial/main amd64 ruby-dev amd64 1:2.3.0+1 [4,408 B]
Fetched 1,360 kB in 3s (452 kB/s)
Selecting previously unselected package libgmpxx4ldbl:amd64.
(Reading database ... 223639 files and directories currently installed.)
Preparing to unpack .../libgmpxx4ldbl_2%3a6.1.0+dfsg-2_amd64.deb ...
Unpacking libgmpxx4ldbl:amd64 (2:6.1.0+dfsg-2) ...
Selecting previously unselected package libgmp-dev:amd64.
Preparing to unpack .../libgmp-dev_2%3a6.1.0+dfsg-2_amd64.deb ...
Unpacking libgmp-dev:amd64 (2:6.1.0+dfsg-2) ...
Selecting previously unselected package ruby2.3-dev:amd64.
Preparing to unpack .../ruby2.3-dev_2.3.1-2~16.04.6_amd64.deb ...
Unpacking ruby2.3-dev:amd64 (2.3.1-2~16.04.6) ...
Selecting previously unselected package ruby-dev:amd64.
Preparing to unpack .../ruby-dev_1%3a2.3.0+1_amd64.deb ...
Unpacking ruby-dev:amd64 (1:2.3.0+1) ...
Processing triggers for libc-bin (2.23-0ubuntu3) ...
Setting up libgmpxx4ldbl:amd64 (2:6.1.0+dfsg-2) ...
Setting up libgmp-dev:amd64 (2:6.1.0+dfsg-2) ...
Setting up ruby2.3-dev:amd64 (2.3.1-2~16.04.6) ...
Setting up ruby-dev:amd64 (1:2.3.0+1) ...
Processing triggers for libc-bin (2.23-0ubuntu3) ...
mint-18 beef # ./install 
,                              
                                     lkOl                             
                                     od cOc                           
                                     'X,  cOo.                        
                                      cX,   ,dkc.                     
                                       ;Kd.    ,odo,.                 
                                        .dXl   .  .:xkl'              
                                          'OKc  .;c'  ,oOk:           
                                            ,kKo. .cOkc. .lOk:.       
                                              .dXx.  :KWKo. 'dXd.     
                                                .oXx.  cXWW0c..dXd.   
                                                  oW0   .OWWWNd.'KK.  
                                          ....,;lkNWx     KWWWWX:'XK. 
 ,o:,                          .,:odkO00XNK0Okxdlc,.     .KWWWWWWddWd 
  K::Ol                   .:d0NXK0OkxdoxO'             .lXWWWWWWWWKW0 
  od  d0.              .l0NKOxdooooooox0.        .,cdOXWWWWWWWWWWWWWx 
  :O   ;K;           ;kN0kooooooooooooK:  .':ok0NWWWWWWWWWWWWWWWWWWK. 
  'X    .Kl        ;KNOdooooooooooooooXkkXWWWWWWWWWWWWWWWWWWWWWWWNd.  
  .N. o. .Kl     'OW0doooooooooooooodkXWWWWWWWWWWWWWWWWWWWWWWWW0l.    
   0l oK' .kO:';kNNkoooooooooooook0XWWWWWWWWWWWWWWWWWWWWWWWKx:.       
   lX.,WN:  .:c:xWkoooooooooood0NWW0OWWWWWWWWWWWWWWWWWWWKo.           
    0O.0WWk'   .XKoooooooooooONWWNo  dWWWWWWWWWWWWWWWWWl              
     oKkNWWWX00NWXdooooooooxXWWNk'   dWWWWWWWWWWWWWWWWX               
      .cONWWWWWWWWOoooooooONWWK:...c0WWWWWWWWWWWWWWWWWW:              
         .;oONWWWWxooooodKWWWWWWWWWWWWWWWWWWWWWWWWWWWWWX.             
              'XW0oooookNWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWd             
              oW0ooooo0WWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWO             
             ;NXdooodKWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWx             
          ;xkOOdooooxOO0KNWWWWWWWWWWWWWWWWWWWWWWWWWWWWWX.             
         .NOoddxkkkkxxdoookKWWWWWWWWWWWWWWWWWWWWWWWWWWX'              
          :KNWWWWWWWWWWX0xooONWWWWWWWWWWWWWWWWWWWWWWWk.               
         .xNXxKWWWWWWWOXWWXxoKWWWWWWWWWWWWWWWWWWWWNk'                 
         OWl cNWWWWWWWk oNWNxKWWWWWWWWWWWWWWWWWNOl.                   
        ,Wk  xWWWWWWWWd  xWWNWWWWWWWWWWWWXOdc,.                       
        .N0   lOXNX0x;  .KWWWWWWWWWWWNkc.                             
         :NO,         'lXWWWWWWWWWNk:.                                
          .dXN0OkxkO0NWWWWWWWWWWKl.                                   
             .';o0WWWWWWWWWWWNk;                                      
                  .cxOKXKKOd;.                                        
                                                                      

#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#
                   -- [ BeEF Installer ] --                      
#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#

[WARN]   This script will install BeEF and its required dependencies (including operating system packages).
Are you sure you wish to continue (Y/n)? 
[INFO]   Detecting OS...
[INFO]   Operating System: Linux
[INFO]   Launching Linux install...
[INFO]   Detecting Linux OS distribution...
[INFO]   OS Distribution: Debian
[INFO]   Installing Debian prerequisite packages...
Ign:1 http://packages.linuxmint.com sarah InRelease                                              
Hit:2 http://security.ubuntu.com/ubuntu xenial-security InRelease                                
Hit:3 http://archive.canonical.com/ubuntu xenial InRelease                                       
Hit:4 http://packages.linuxmint.com sarah Release                                     
Hit:6 http://archive.ubuntu.com/ubuntu xenial InRelease         
Get:7 http://archive.ubuntu.com/ubuntu xenial-updates InRelease [102 kB]
Hit:8 http://archive.ubuntu.com/ubuntu xenial-backports InRelease
Fetched 102 kB in 12s (8,164 B/s)                                                                
Reading package lists... Done
Reading package lists... Done
Building dependency tree       
Reading state information... Done
autoconf is already the newest version (2.69-9).
automake is already the newest version (1:1.15-4ubuntu1).
bison is already the newest version (2:3.0.4.dfsg-1).
build-essential is already the newest version (12.1ubuntu2).
libncurses5-dev is already the newest version (6.0+20160213-1ubuntu1).
libreadline6-dev is already the newest version (6.3-8ubuntu2).
libsqlite3-0 is already the newest version (3.11.0-1ubuntu1).
libsqlite3-dev is already the newest version (3.11.0-1ubuntu1).
libtool is already the newest version (2.4.6-0.1).
libyaml-dev is already the newest version (0.1.6-3).
sqlite3 is already the newest version (3.11.0-1ubuntu1).
curl is already the newest version (7.47.0-1ubuntu2.6).
git is already the newest version (1:2.7.4-0ubuntu1.3).
libc6-dev is already the newest version (2.23-0ubuntu10).
libssl-dev is already the newest version (1.0.2g-1ubuntu4.10).
libxml2-dev is already the newest version (2.9.3+dfsg1-1ubuntu0.5).
libxslt1-dev is already the newest version (1.1.28-2.1ubuntu0.1).
openssl is already the newest version (1.0.2g-1ubuntu4.10).
subversion is already the newest version (1.9.3-2ubuntu1.1).
zlib1g is already the newest version (1:1.2.8.dfsg-2ubuntu4.1).
zlib1g-dev is already the newest version (1:1.2.8.dfsg-2ubuntu4.1).
nodejs is already the newest version (4.2.6~dfsg-1ubuntu4.1).
0 to upgrade, 0 to newly install, 0 to remove and 720 not to upgrade.
[INFO]   Detecting Ruby environment...
[INFO]   Ruby version 2.3.1 is installed
[INFO]   Detecting bundler gem...
[INFO]   bundler gem is installed
Installing required Ruby gems...
Don't run Bundler as root. Bundler can ask for sudo if it is needed, and installing your bundle as
root will break this application for all non-root users on this machine.
Fetching gem metadata from https://rubygems.org/...........
Fetching gem metadata from https://rubygems.org/..
Resolving dependencies...
Using rake 12.1.0
Using public_suffix 3.0.0
Using addressable 2.5.2
Using ansi 1.5.0
Using buftok 0.2.0
Using bundler 1.16.1
Using chunky_png 1.3.8
Using daemons 1.2.4
Using data_objects 0.10.17
Using dm-core 1.2.1
Using dm-do-adapter 1.2.0
Using dm-migrations 1.2.0
Using fastercsv 1.5.5
Fetching json 1.8.6
Installing json 1.8.6 with native extensions
Fetching json_pure 1.8.6
Installing json_pure 1.8.6
Fetching multi_json 1.12.2
Installing multi_json 1.12.2
Fetching dm-serializer 1.2.2
Installing dm-serializer 1.2.2
Fetching do_sqlite3 0.10.17
Installing do_sqlite3 0.10.17 with native extensions
Fetching dm-sqlite-adapter 1.2.0
Installing dm-sqlite-adapter 1.2.0
Fetching unf_ext 0.0.7.4
Installing unf_ext 0.0.7.4 with native extensions
Fetching unf 0.1.4
Installing unf 0.1.4
Fetching domain_name 0.5.20170404
Installing domain_name 0.5.20170404
Fetching eventmachine 1.0.9.1
Installing eventmachine 1.0.9.1 with native extensions
Fetching http_parser.rb 0.6.0
Installing http_parser.rb 0.6.0 with native extensions
Fetching em-websocket 0.5.1
Installing em-websocket 0.5.1
Fetching equalizer 0.0.11
Installing equalizer 0.0.11
Fetching erubis 2.7.0
Installing erubis 2.7.0
Fetching espeak-ruby 1.0.4
Installing espeak-ruby 1.0.4
Fetching execjs 2.7.0
Installing execjs 2.7.0
Fetching filesize 0.1.1
Installing filesize 0.1.1
Fetching geoip 1.6.3
Installing geoip 1.6.3
Fetching http-cookie 1.0.3
Installing http-cookie 1.0.3
Fetching http-form_data 2.0.0
Installing http-form_data 2.0.0
Fetching http 3.0.0
Installing http 3.0.0
Fetching rkelly-remix 0.0.7
Installing rkelly-remix 0.0.7
Fetching jsobfu 0.4.2
Installing jsobfu 0.4.2
Fetching libv8 3.16.14.19 (x86_64-linux)
Installing libv8 3.16.14.19 (x86_64-linux)
Fetching thread_safe 0.3.6
Installing thread_safe 0.3.6
Fetching memoizable 0.4.2
Installing memoizable 0.4.2
Fetching metasm 1.0.3
Installing metasm 1.0.3
Fetching mime-types-data 3.2016.0521
Installing mime-types-data 3.2016.0521
Fetching mime-types 3.1
Installing mime-types 3.1
Fetching mini_portile2 2.3.0
Installing mini_portile2 2.3.0
Fetching mojo_magick 0.5.6
Installing mojo_magick 0.5.6
Fetching msgpack 1.1.0
Installing msgpack 1.1.0 with native extensions
Fetching nokogiri 1.8.1
Installing nokogiri 1.8.1 with native extensions
Fetching rb-readline 0.5.5
Installing rb-readline 0.5.5
Fetching robots 0.10.1
Installing robots 0.10.1
Fetching rex 2.0.11
Installing rex 2.0.11
Fetching msfrpc-client 1.1.1
Installing msfrpc-client 1.1.1
Fetching multipart-post 2.0.0
Installing multipart-post 2.0.0
Fetching naught 1.1.0
Installing naught 1.1.0
Fetching netrc 0.11.0
Installing netrc 0.11.0
Fetching parseconfig 1.0.8
Installing parseconfig 1.0.8
Fetching rqrcode 0.10.1
Installing rqrcode 0.10.1
Fetching qr4r 0.4.1
Installing qr4r 0.4.1
Fetching rack 1.6.8
Installing rack 1.6.8
Fetching rack-protection 1.5.3
Installing rack-protection 1.5.3
Fetching rainbow 2.2.2
Installing rainbow 2.2.2 with native extensions
Fetching ref 2.0.0
Installing ref 2.0.0
Fetching rest-client 2.0.2
Installing rest-client 2.0.2
Fetching rexec 1.6.3
Installing rexec 1.6.3
Fetching rubydns 0.7.3
Installing rubydns 0.7.3
Fetching rubyzip 1.2.1
Installing rubyzip 1.2.1
Fetching rushover 0.3.0
Installing rushover 0.3.0
Fetching simple_oauth 0.3.1
Installing simple_oauth 0.3.1
Fetching tilt 2.0.8
Installing tilt 2.0.8
Fetching sinatra 1.4.8
Installing sinatra 1.4.8
Fetching slack-notifier 2.3.1
Installing slack-notifier 2.3.1
Fetching tins 1.15.0
Installing tins 1.15.0
Fetching term-ansicolor 1.6.0
Installing term-ansicolor 1.6.0
Fetching therubyracer 0.12.2
Installing therubyracer 0.12.2 with native extensions
Fetching thin 1.7.2
Installing thin 1.7.2 with native extensions
Fetching twitter 6.2.0
Installing twitter 6.2.0
Fetching uglifier 3.2.0
Installing uglifier 3.2.0
Bundle complete! 30 Gemfile dependencies, 75 gems now installed.
Gems in the groups test and development were not installed.
Use `bundle info [gemname]` to see where a bundled gem is installed.
Post-install message from mojo_magick:

Thanks for installing MojoMagick - keepin it simple!

*** To make this gem work, you need a few binaries!
Make sure you've got ImageMagick available.  http://imagemagick.org
If you plan to build images with text (using the "label" method) you'll need freetype and ghostscript as well.
Check out http://www.freetype.org and http://ghostscript.com respectively for installation info.


==========================================

[INFO]   Install completed successfully!
[INFO]   Run './beef' to launch BeEF

==========================================

mint-18 beef # ./beef 
[ 0:05:29][*] Browser Exploitation Framework (BeEF) 0.4.7.0-alpha
[ 0:05:29]    |   Twit: @beefproject
[ 0:05:29]    |   Site: http://beefproject.com
[ 0:05:29]    |   Blog: http://blog.beefproject.com
[ 0:05:29]    |_  Wiki: https://github.com/beefproject/beef/wiki
[ 0:05:29][*] Project Creator: Wade Alcorn (@WadeAlcorn)
[ 0:05:30][*] BeEF is loading. Wait a few seconds...
[ 0:05:33][*] 8 extensions enabled.
[ 0:05:33][*] 301 modules enabled.
[ 0:05:33][*] 2 network interfaces were detected.
[ 0:05:33][+] running on network interface: 127.0.0.1
[ 0:05:33]    |   Hook URL: http://127.0.0.1:3000/hook.js
[ 0:05:33]    |_  UI URL:   http://127.0.0.1:3000/ui/panel
[ 0:05:33][+] running on network interface: 172.16.191.189
[ 0:05:33]    |   Hook URL: http://172.16.191.189:3000/hook.js
[ 0:05:33]    |_  UI URL:   http://172.16.191.189:3000/ui/panel
[ 0:05:33][!] Warning: Default username and weak password in use!
[ 0:05:33]    |_  New password for this instance: e746ffa1f2073ecb16603fc49fb1e58c
[ 0:05:33][*] RESTful API key: 090d6bece33df7b363045b1a3e8a66294be0f6ef
[ 0:05:33][*] HTTP Proxy: http://127.0.0.1:6789
[ 0:05:33][*] BeEF server started (press control+c to stop)
[22:52:41][*] New Hooked Browser [id:1, ip:127.0.0.1, browser:FF-47, os:Linux-], hooked domain [127.0.0.1:3000]
^Cmint-18 beef # 
mint-18 beef # 
mint-18 beef # ruby --version
ruby 2.3.1p112 (2016-04-26) [x86_64-linux-gnu]

```
